### PR TITLE
DRY analyzer specs

### PR DESCRIPTION
**Pull request by wfleming**
These pieces were repeated pretty much verbatim between the different
analyzer specs. That seems especially silly since this is, after all, an
engine for pointing out code duplication.

:eyes: @codeclimate/review

### CODE REVIEW

**Comment by brynary** [comment]
This looks good, but did we consider moving these to use the new `codeclimate test` harness?

File: spec/support/analyzer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module AnalyzerSpec
+  def self.included(base)
```
Comment (pbrisbin) [inline spec/support/analyzer_spec.rb]: Would this be better added via `RSpec.configure`? Maybe on a `in_tmpdir: true` tag or something?

**Comment by wfleming** [comment]
@brynary That's a good idea, but not something I want to do just now: these specs test some other things like readup contents that I don't believe the test harness handles? It would be good to use the test harness & break the other things into different specs, but it's a bigger refactor than I want to get into right now.

**Comment by pbrisbin** [comment]
LGTM

**Pull request merged**